diesel_derives2/src/as_changeset.rs

```diff
@@ -0,0 +1,91 @@
+use quote;
+use syn;
+
+use field::*;
+use meta::*;
+use model::*;
+use util::*;
+
+pub fn derive(item: syn::DeriveInput) -> quote::Tokens {
+    let treat_none_as_null = MetaItem::with_name(&item.attrs, "changeset_options")
+        .map(|meta| {
+            meta.nested_item("treat_none_as_null")
+                .required()
+                .expect_bool_value()
+        })
+        .unwrap_or(false);
+    let model = Model::from_item(&item);
+    let struct_name = &model.name;
+    let table_name = model.table_name();
+
+    let (_, ty_generics, where_clause) = item.generics.split_for_impl();
+    let mut impl_generics = item.generics.clone();
+    impl_generics.params.push(parse_quote!('update));
```

Comment:
You are going to want another split_for_impl after this line.

```rust
let (impl_generics, _, _) = impl_generics.split_for_impl();
```

Otherwise it does not work for types that have generic type parameter defaults.

```rust
#[derive(AsChangeset)]
#[table_name = "users"]
#[primary_key(name)]
struct Changes2<T = String> {
    name: T,
    hair_color: String,
}
```

```
error: defaults for type parameters are only allowed in `struct`, `enum`, `type`, or `trait` definitions.
   --> tests/update.rs:390:14
    |
390 |     #[derive(AsChangeset)]
    |              ^^^^^^^^^^^
    |
    = note: #[deny(invalid_type_param_default)] on by default
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
    = note: for more information, see issue #36887 <https://github.com/rust-lang/rust/issues/36887>
```

Reply:
This code doesn't currently work with generic structs, but good catch.